src/utils/FlowUtils.ts

```diff
@@ -30,7 +31,13 @@ export function validateFillForDeposit(
   // Note: this short circuits when a key is found where the comparison doesn't match.
   // TODO: if we turn on "strict" in the tsconfig, the elements of FILL_DEPOSIT_COMPARISON_KEYS will be automatically
   // validated against the fields in Fill and Deposit, generating an error if there is a discrepency.
-  const invalidKey = RELAYDATA_KEYS.find((key) => relayData[key].toString() !== deposit[key].toString());
+  let invalidKey = RELAYDATA_KEYS.find((key) => relayData[key].toString() !== deposit[key].toString());
+
+  // There should be no paths for `messageHash` to be unset, but mask it off anyway.
+  // @todo Add test.
```

Comment:
Can drop this todo now: https://github.com/across-protocol/sdk/pull/850/commits/db670655dd3918ebcb630d004ee8d9143d9a5f1b

```suggestion
```